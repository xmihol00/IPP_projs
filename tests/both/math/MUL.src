
.IPPcode21

DEFVAR GF@num
DEFVAR GF@int
MOVE GF@int int@2

MUL GF@num int@0 int@15
WRITE GF@num
WRITE string@\010

MUL GF@num GF@int int@12
WRITE GF@num
WRITE string@\010

MUL GF@num int@-3 GF@int
WRITE GF@num
WRITE string@\010

MUL GF@num GF@num GF@int
WRITE GF@num
WRITE string@\010

MUL GF@num GF@num GF@num
WRITE GF@num
WRITE string@\010

WRITE string@\010

CREATEFRAME
DEFVAR TF@_
DEFVAR TF@!
MOVE TF@_ int@9

MUL TF@! TF@_ TF@_
WRITE TF@!
WRITE string@\010

MUL TF@! TF@! int@-2
WRITE TF@!
WRITE string@\010

MUL TF@_ int@2 TF@!
WRITE TF@_
WRITE string@\010

MUL TF@_ int@5 int@6
WRITE TF@_
WRITE string@\010

WRITE string@\010

PUSHFRAME
WRITE LF@_
WRITE string@\010
WRITE LF@!
WRITE string@\010

MUL LF@_ LF@_ int@-2
WRITE LF@_
WRITE string@\010

MUL LF@! int@-4 LF@!
WRITE LF@!
WRITE string@\010

MUL LF@! LF@_ LF@!
WRITE LF@!
WRITE string@\010

MUL LF@_ int@5 int@-5
WRITE LF@_
WRITE string@\010
WRITE string@\010

POPFRAME
WRITE TF@_
WRITE string@\010
WRITE TF@!
WRITE string@\010
