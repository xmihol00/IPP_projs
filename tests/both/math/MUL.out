0
24
-6
-12
144

81
-162
-324
30

30
-162
-60
648
-38880
-25

-25
-38880
